includes/admin/controllers/class-admin-listings.php

```diff
@@ -700,6 +702,32 @@ public function ajax_clear_payment_history() {
         wp_send_json_success( array( 'message' => _x( 'Listing\'s payment history successfully deleted', 'admin listings', 'business-directory-plugin' ) ) );
     }
 
+	/**
+	 * Assign a plan to a listing.
+	 * This assigns a plan to a listing in the admin backend.
+	 *
+	 * @since x.x
+	 */
+	public function ajax_assign_plan_to_listing() {
+		check_ajax_referer( 'wpbdp_ajax', 'nonce' );
```

Comment:
Looks like we're missing the permission check here.

Reply:
Permission check added to match metabox check